### DISCUSSION TITLE
Is there anyway to load a child's relationship as a relation in the parent?

### DISCUSSION THREAD

**Discussion by scajal** [laravel/framework#46900] [Q&A]
I have the following scenario:

```
orders
    id - integer

coins
    id - integer

order_lines
    id - integer
    type - string
    order_id - integer
    coin_id - integer    
```

Is there any way I can load the coin of a line as a relation in the orders? The idea is to do `$order->coin()` (similar to what `hasOneThrough` does). 
I'd also need to filter the lines from which i'd get the coin, for example, get the coin from the first line with type "A" (`$order->coinA()`).
I want it to be a relationship since I need it to be lazy loaded.

Thanks!

**Comment by henzeb** [comment]
See hasManyThrough https://laravel.com/docs/10.x/eloquent-relationships#has-many-through

**Reply by scajal** [reply]
Thanks, I know it exists, but is it possible to filter the query it makes in the intermediate model?

**Reply by henzeb** [reply]
Yes, you can chain any query builder methods to filter out anything you don't want.

**Reply by scajal** [reply]
Yes, but the filters will be applied to the last model (in this case, the coin), **not** to the intermediate model, which is what I need.

**Reply by henzeb** [reply]
You need to specify the table name in your where  clause. Where('order_lines.type', 'A').

**Reply by scajal** [reply]
Not working. Laravel uses the connection field in Coin, not in OrderLine.

**Reply by Angel5a** [reply]
Define auxiliary relation coinA in order_lines model, and use it as a thru path.
